src/pal/src/loader/module.cpp

```diff
@@ -1628,6 +1636,12 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     LockModuleList();
 
     dl_handle = LOADLoadLibraryDirect(shortAsciiName);
+#ifdef __APPLE__
+    if (!dl_handle && isLibcRequested)
+    {
+        dl_handle = LOADLoadLibraryDirect("/usr/lib/libc.dylib");
```

Comment:
@jkotas, I only have El Capitan to test this patch hence used the harmless fallback approach. If this is a known fact that all supported macOS versions will have `/usr/lib/libc.dylibc` as valid path, then I agree we should use just one path. Once confirmed, I will make the changed.

@josteink, in my second commit (7db7794), I fixed the outdated comment about libc. I can put one here as well. :) On macsOS and BSDs and Alpine, there is no equivalent of `#include <gnu/lib-names.h>` unfortunately.